### PR TITLE
Adjusting recipes using electric motors to use correct motor size relative to resulting item sizes

### PR DESCRIPTION
**Pull request by stubkan**
#### Summary
Balance "Adjusting recipes using electric motors to use correct motor size relative to resulting item sizes"

#### Purpose of change

As part 2 of my earlier motor volume/weight rebalancing (#64482) - this PR is to deal with electric motor recipes as there are a lot to change, since many of them appear to be using the wrong electric motor tier in their recipes. Such as;

* Electric hedge trimmer using a tiny motor that would be more suitable in an electric toothbrush.
* RC car using an electric motor that was basically the size of the RC car itself (a motor capable of propelling a bicycle)
* Angle grinder tool was using a motor that was big enough to power a go-kart or golf cart.

It doesn't help that motors that are actually quite large are named 'small'.

#### Describe the solution

Collected a list of recipes (crafting, disassembly, bashing, deconstruction) that used electric motors.  Grabbed reference pictures that reflected the size of each electric motor tier that the game uses (micro, tiny, small, normal, large) and went through each recipe, making a table.  

Each item that didnt have a motor that was the right size was marked.  For items that I was unsure about, a google image, such as "washing machine motor" would provide a picture of a man holding a washing machine motor, making it an easy reference point.  After the list was done, I went through the JSON, changing each recipe to make the motor component the correct size.

#### Describe alternatives you've considered

None.

#### Testing

Ran and compiled the game with updated JSON, loaded a world, examined items ingame..  No errors so far.

#### Additional context

I believe the wrong recipes may be in part due to the smallest electric motor tier being 'micro motor' - and over time the people creating recipes searched for 'electric motor' and the lowest one they found was the tiny electric motor, and likely missed that the micro motor existed.  Also the descriptions for electric motors were vague about their power and tier position. I have renamed the micro motor to 'micro electric motor', so it matches all the other electric motor tier naming.  Better descriptions to hopefully more clearly describe the motors power tier are in as well.

The recipes in this game are an absolute morass of unchecked and incorrect correspondences, I tried to keep within the scope of the PR, but a few recipes stood out, so I made some small changes - in relation to motor sizes and recipe amounts, such as Exodii robots, robot components etc using overly large or too many motors to make the resulting robot contents make sense.

Tweaked crafting recipes for tiny/small motors (which are vehicle motors) to require welding, and more sensible components.

Here is a changelog, along with reference images to help visualize how each motor tier is sized.

# *MICRO MOTORS*
image

| micro motor 125g | electric toothbrush or RC car |
| -- | -- |
| food processor | changed to tiny motor |
| hedge trimmer | changed to tiny motor |

# *TINY MOTORS*
image

| Tiny motor 1.5 kg | bicycle or power tool |
| -- | -- |
| condensor coil | changed to micro motor |
| electric carver | changed to micro motor |
| hand-crank charger | changed to micro motor |
| minifreezer | changed to micro motor |
| minifridge | changed to micro motor |
| RC car | changed to micro motor |
| small closed loop extractor | changed to micro motor |
| small vacuum oven | changed to micro motor |
| set of android arms | reduced amount |
| set of android legs | reduced amount |
| set of tiny spidery legs | changed to micro motor |
| small 1-cylinder engine | changed to micro motor |
| anesthetic machine | changed to micro motor |
| dialysis machine | changed to micro motor |

# *SMALL MOTORS*
image

| Small motor 9 kg | motorbike or golf cart | 
| -- | -- |
| 12-gauge gatling gun | changed to tiny motor |
| angle grinder | changed to tiny motor |
| circular saw | changed to tiny motor |
| concrete mixer | changed to tiny motor |
| disconnected freezer | changed to tiny motor |
| disconnected refrigerator | changed to tiny motor |
| electric chainsaw (off) | changed to tiny motor |
| electric oil press | changed to tiny motor |
| washing machine | changed to tiny motor |
| 1-cylinder engine | changed to micro motor |
| blade trap | changed to tiny motor |
| disconnected autoclave | changed to tiny motor |
| disconnected large water purifier | changed to tiny motor |
| disconnected refrigerator | changed to small motor |
| electric polisher | changed to tiny motor |
| household dishwasher | changed to tiny motor |
| household washing machine | changed to tiny motor |
| I4 diesel engine | changed to tiny motor |
| I4 engine | changed to tiny motor |
| I6 diesel engine | changed to tiny motor |
| large 1-cylinder engine | changed to tiny motor |
| Microcentrifuge | changed to micro motor |
| set of omni wheels | changed to tiny motor |
| set of reverse-jointed legs | changed to tiny motor |
| set of spidery legs | changed to tiny motor
| V12 diesel engine | changed to tiny motor |
| V12 engine | changed to tiny motor |
| V2 engine | changed to tiny motor |
| V6 diesel engine | changed to tiny motor |
| V6 engine | changed to tiny motor |
| V8 diesel engine | changed to tiny motor |
| V8 engine | changed to tiny motor |
| furn broken generator | changed to tiny motor |
| central air filter | changed to tiny motor |
| Centrifuge | changed to tiny motor |
| chemical mixer | changed to tiny motor |
| cooling unit | changed to tiny motor |
| Dryer | changed to tiny motor |
| filled autoclave | changed to tiny motor |
| fume hood | changed to tiny motor |
| grid microwave | changed to micro motor |
| mounted electric oil press | changed to tiny motor |

# *MOTORS*
image

| motor 46 kg | 100HP electric car |
| -- | -- |
| broken Exodii quadruped | changed to small motor |
| broken Exodii turret | changed to small motor |
| broken Exodii worker | changed to small motor |
| broken prototype robot | changed to small motor |
| disconnected air compressor | changed to small motor |
| document shredder | changed to tiny motor |
| electronic machinery | changed to tiny motor |
| large satellite dish | changed to small motor |
| light machinery | changed to small motor |
| robotic arm | changed to small motor |
| electronic machinery | changed to tiny motor |
| light machinery | changed to small motor |
| nanofabricator | changed to small motor |
| robotic arm | changed to small motor |

### CODE REVIEW

**Comment by stubkan** [comment]
Updated crafting recipes for creating the tiny and small motors (bear in mind these are powerful motors that are capable of propelling vehicles)

Used to only require trivial materials, soldering iron, 3 copper wire, electronic scrap, a steel chunk and you could have a vehicle motor.  Changed that to require actual welding, ball bearings, pipe, bigger steel amounts.  The larger small motor requires small steel sheets as well.

image

**Review by Maleclypse — CHANGES_REQUESTED** [review]
Once the changes have been made to show where the new engine stats come from on the other PR please ping me in discord and I'll review this PR.

**Comment by stubkan** [comment]
I am unsure what you are requesting

**Comment by stubkan** [comment]
Electric motors (especially in vehicles) have a different power scale, perhaps due to there being fewer motors available.  

As you can see from the table below, electric motor powers skip two combustion engine tiers, each time they go up a tier.

| COMBUSION | POWER | ELECTRIC | POWER |
| -- | -- | micro | -- |
| 25cc | 3728 W | tiny | 800 W |
| 50cc | 7370 W | small | 7040 W |
| 200cc | 20500 W | normal | 36000 W |
| 600cc | 37300 | large | 150000 W |
| 1600cc | 93250 W | enhanced | 282000 W |
| V 6 | 150000 W | super | 475000 W |
| V 8 | 282000 W | -- | -- |
| V 12 | 475000 W | -- | -- |

I will update the descriptions, so the power and size are more clear in them, ingame - basing these off the engine power output and type of vehicle they are installed in.

I'd like to increase the power on the electric bicycle motor (tiny) from 800W to at least 2000W, I believe the same reason many people put the tiny into far smaller tools may be the same reason it has such a low power rating.  But this change might wait until the Part 1 PR is merged, so the volume/weight changes from that carry over first.

**Comment by stubkan** [comment]
Since all the electric motor item descriptions are quite vague as to their power and place on the electric motor tier, their descriptions are updated.  Here is the changelog.

    motor_micro
    OLD : "A very small electric motor like those used in RC cars. Useful in lots of electronics recipes."
    NEW : "A very small electric motor like those used in RC cars or electric hair trimmers.",

    motor_tiny ( DOUBLED VALUES, power to 1600 W, energy_consumption to 2 kW )
    OLD : "A tiny electric motor.  Useful for crafting.",
    NEW : "A small electric motor, suitable for powering a bicycle or a hedge trimmer.",

    motor_small
    OLD : "A small electric motor.  Useful for crafting.",
    NEW : "A light-duty electric motor.  Strong enough for an electric moped or golf cart.",

    motor
    OLD : "A powerful electric motor. Useful for crafting.",
    NEW : "A standard electric motor.  Suitable for use in ultra-light vehicles.",

    motor_large
    OLD : "A large, very powerful electric motor.  Useful for crafting.",
    NEW : "A large electric motor.  Could be used to power a full-size car or SUV.",

    motor_enhanced
    OLD : "A very powerful, yet lightweight electric motor.  Useful for crafting.",
    NEW : "A powerful electric motor, as found in those consumer electric sports cars with ridiculous or ludicrous acceleration options.",

    motor_super
    OLD : "The most powerful electric motor on the market.  Useful for crafting.",
    NEW : "An extremely powerful electric motor.  This delivers massive torque and would be well suited to an electric racing car.",

File: data/json/furniture_and_terrain/furniture-appliances.json
Comment (Maleclypse) [inline data/json/furniture_and_terrain/furniture-appliances.json:549]: So if I'm reading this correctly the paper shredder has a 3lb motor?  I'll come back to this later today (hopefully) to keep reviewing.

Comment (stubkan) [reply data/json/furniture_and_terrain/furniture-appliances.json:549]: Yeah, it WAS a 100 lb / 46 kg motor.  That one was way overkill.

image

Comment (Maleclypse) [reply data/json/furniture_and_terrain/furniture-appliances.json:549]: I googled what paper shredder for home offices weigh and it's 8-9 lbs and since it's basically motor, blades and trash can 1/3rd being motor makes sense.

File: data/json/items/vehicle/motors.json
Comment (Maleclypse) [inline data/json/items/vehicle/motors.json:71]: https://www.amazon.com/Electric-Brushed-2750RPM-Scooter-Control/dp/B07KVVL1NM/ref=asc_df_B07KVVL1NM/?tag=hyprod-20&linkCode=df0&hvadid=317203115653&hvpos=&hvnetw=g&hvrand=14448846602955151091&hvpone=&hvptwo=&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=1026082&hvtargid=pla-635620784133&psc=1

The bike motor above is 4lbs so I believe mopeds and electric bicycles are probably more similar than they are different.

File: data/json/recipes/other/parts.json
Comment (Maleclypse) [inline data/json/recipes/other/parts.json:419]:
```suggestion
    "components": [ [ [ "cable", 5 ] ], [ [ "e_scrap", 3 ] ], [ [ "scrap", 1 ] ], [ [ "bearing", 4 ] ] ]
```
The dda pipe is significantly longer and larger than these motors I believe.

Comment (Maleclypse) [reply data/json/recipes/other/parts.json:419]: I guess my issue is that maybe you could use a pipe to make this motor but you'd be unable to reverse it and get the pipe back

**Review by Maleclypse — APPROVED** [review]
I think if you address or contest the changes I've suggested this will be in a good place.

File: data/json/items/vehicle/motors.json
Comment (stubkan) [reply data/json/items/vehicle/motors.json:71]: That motor you linked is for one of them stand on e-scooters kids are using, which is more in line with a bicycle sized vehicle... 

When I think of a moped or scooter, I think of a vehicle which has the equivalent to a 50cc engine - like this one
image

Perhaps I should change 'moped' on the bigger motor description to something thats more clearly higher tier, since the word can also mean a very small almost-bicycle in some places

File: data/json/recipes/other/parts.json
Comment (stubkan) [reply data/json/recipes/other/parts.json:419]: That makes sense

File: data/json/items/vehicle/motors.json
Comment (stubkan) [reply data/json/items/vehicle/motors.json:71]: https://www.youtube.com/watch?v=5QYyzVyWckw  This guy shows 3 different motors which are the golf cart / 50cc bike equivalent.

**Pull request merged**
